### DISCUSSION TITLE
The Ultimate Administrative Conundrum

### DISCUSSION THREAD

**Discussion by kovitikus** [evennia/evennia#2511] [General Discussion]
Scripting. Botting. Multiple accounts. Should they be allowed? Should they be banned? What are the consequences?

#### Scripting
As anyone who plays a text game might know, scripting them is extremely easy to do. You setup some conditions and triggers and evaluate the situation to determine what input to execute. Scripting is impossible to fight against in a text-based game. It's simply too easy to design responses, because the text players are expecting to read is generally static, which is good design. I've watched a MUD attempt to combat it for over 20 years, only to eventually give up and fully accept it as being an inevitability.

For my project, I think scripting is a no-brainer. Not only will it be allowed and welcomed, but I'll do my best to provide in-game tools to help people script actions. This will especially be helpful for those who aren't able to write their own scripts in languages like Lua, keeping things a level playing field.

I'd also like to point out that repetitive stress injury is real and people already spend too much time at the keyboard as it is. Scripting helps reduce the amount of typing players have to do and reduces their physical stress.

#### Botting
Here's where most games will nope out. They go by the philosophy that while scripting is allowed, you should be at your keyboard and responsive at all times. Botting is the act of someone going AFK, even going to sleep or out to the store, while the game plays itself. This is heavily frowned upon and generally outright banned.

My problem with banning botting is that you are unlikely to stop people from doing it. Keeping people from botting requires real-time administrative overhead. It's not generally the type of thing you can automate checks for.

But also, the botting checks I've seen are immersion breaking. Asking a player to stop what they are doing so they can solve a math problem or some other human question they must answer. Players that aren't botting feel punished and annoyed when asked to prove they are at their keyboard. Imagine being randomly stopped to solve a CAPTCHA `(Completely Automated Public Turing test to tell Computers and Humans Apart)` while you are enjoying your game.

Do you have any possible solutions for this? Ones that wouldn't break immersion, wouldn't irritate players, and would cut out the administrative overhead?

Despite all efforts to stop botting, it's still going to happen. I know, because I have botted in MUDs quite a bit myself and only once was ever "caught", which I quickly talked my way out of.

#### Multiple Accounts
In many online games, there are strict rules about playing multiple accounts at the same time. World of Warcraft -used- to allow it for the past 15 years. It's only recently that they finally started to crack down on multi-boxers `(a term used to describe players that run multiple computers to play multiple accounts)`.

In MUDs, as far as the ones I've encountered, playing multiple accounts simultaneously is a bannable offense. When you combine scripting, and botting especially, with multiple accounts you can have one player pretty much break the game.

A player botting multiple accounts can easily become a jack-of-all-trades. One account trains only combat, another account trains only crafting, etc. Between their characters, they can provide themselves with all of the services they could ever want and never charge themselves any fees. This not only breaks the social aspect (likely not role-playing between all of their characters) of a MUD and the need for players to seek out other players for services, but it also breaks the economy.

At the same time, can this really be prevented? I've played and botted multiple accounts myself on a MUD and it was extremely easy to bypass the checks in place. All I had to do was find a proxy service that allowed me to buy a dozen or so IP addresses and then set up various clients with different proxy information. There's even 3rd party applications, such as Proxifier, that allows you to specify a program only connect through a proxy; allowing you to proxy an application that has no proxy settings built-in.

So long as I didn't do anything stupid, like have all of the accounts always playing with each other at all times and logging in at the same time or logging out at the same time, it was extremely easy to script or bot multiple accounts at once.

#### Conclusion
I don't have a solution for any of these issues, without giving myself a huge headache and wasting administrative time chasing down "cheaters". But at the same time, allowing this stuff to be rampant and legal causes a lot of issues.

Are these issues really important in the long run?

If I don't allow botting and multiple accounts, the only people who will suffer are those who want to follow the rules in the first place. There will be always be players who are really good at this stuff and will simply bypass my rules anyways. Allowing the rule-breakers to have the advantage and punishing those willing to follow the rules only serves to leave a gap between the two types of players, making it unfair in my opinion.

***

Reddit Thread: https://old.reddit.com/r/MUD/comments/pbmlb9/the_ultimate_administrative_conundrum/

**Comment by jaborsh** [comment]
As someone that has chronically scripted/botted/cheated in basically every game I've played, I'd say the best thing an Administrator can do is design a system that doesn't rely on so many tasks that are convenient to automate.

For example: if you want to have a fishing system, just have the fishing run autonomously so that the character is free to interact with other players or do other things in the meantime instead of type/script monotonous commands at pre-defined intervals.

**Reply by kovitikus** [reply]
Yes. I'll definitely be designing my systems with those sorts of considerations in mind.

I feel like one of the problems I run into with MUDs is that the character is treated like a mindless entity you have to explicitly execute every action for it. My philosophy is that the character has agency. It knows how to do something. There's no reason to force the player to manually complete every step for the character. My design leans toward the player submitting meta commands to the character.

> make 100 torches

Then the character proceeds to make torches from the surrounding area until the task is complete, including searching for the sticks on the forest floor.
